From: Maxim Dounin <mdounin@mdounin.ru>
Date: Thu, 7 Dec 2017 17:09:33 +0300
Subject: Configure: moved IP_BIND_ADDRESS_NO_PORT test.

In 2c7b488a61fb, IP_BIND_ADDRESS_NO_PORT test was accidentally placed
between SO_BINDANY, IP_TRANSPARENT, and IP_BINDANY tests.  Moved it after
these tests.

--- a/auto/unix
+++ b/auto/unix
@@ -354,42 +354,42 @@ ngx_feature_test="setsockopt(0, SOL_SOCK
 . auto/feature
 
 
-# Linux IP_BIND_ADDRESS_NO_PORT
+# Linux transparent proxying
 
-ngx_feature="IP_BIND_ADDRESS_NO_PORT"
-ngx_feature_name="NGX_HAVE_IP_BIND_ADDRESS_NO_PORT"
+ngx_feature="IP_TRANSPARENT"
+ngx_feature_name="NGX_HAVE_TRANSPARENT_PROXY"
 ngx_feature_run=no
 ngx_feature_incs="#include <sys/socket.h>
                   #include <netinet/in.h>"
 ngx_feature_path=
 ngx_feature_libs=
-ngx_feature_test="setsockopt(0, IPPROTO_IP, IP_BIND_ADDRESS_NO_PORT, NULL, 0)"
+ngx_feature_test="setsockopt(0, IPPROTO_IP, IP_TRANSPARENT, NULL, 0)"
 . auto/feature
 
 
-# Linux transparent proxying
+# FreeBSD bind to any address for transparent proxying
 
-ngx_feature="IP_TRANSPARENT"
+ngx_feature="IP_BINDANY"
 ngx_feature_name="NGX_HAVE_TRANSPARENT_PROXY"
 ngx_feature_run=no
 ngx_feature_incs="#include <sys/socket.h>
                   #include <netinet/in.h>"
 ngx_feature_path=
 ngx_feature_libs=
-ngx_feature_test="setsockopt(0, IPPROTO_IP, IP_TRANSPARENT, NULL, 0)"
+ngx_feature_test="setsockopt(0, IPPROTO_IP, IP_BINDANY, NULL, 0)"
 . auto/feature
 
 
-# FreeBSD bind to any address for transparent proxying
+# Linux IP_BIND_ADDRESS_NO_PORT
 
-ngx_feature="IP_BINDANY"
-ngx_feature_name="NGX_HAVE_TRANSPARENT_PROXY"
+ngx_feature="IP_BIND_ADDRESS_NO_PORT"
+ngx_feature_name="NGX_HAVE_IP_BIND_ADDRESS_NO_PORT"
 ngx_feature_run=no
 ngx_feature_incs="#include <sys/socket.h>
                   #include <netinet/in.h>"
 ngx_feature_path=
 ngx_feature_libs=
-ngx_feature_test="setsockopt(0, IPPROTO_IP, IP_BINDANY, NULL, 0)"
+ngx_feature_test="setsockopt(0, IPPROTO_IP, IP_BIND_ADDRESS_NO_PORT, NULL, 0)"
 . auto/feature
 
 
